src/mpl/include/mpl_proc_mutex_posix.h

```diff
@@ -0,0 +1,125 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *
+ *  (C) 2001 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ */
+
+/*
+ * Interprocess Mutex
+ */
+#ifndef MPL_PROC_MUTEX_POSIX_H_INCLUDED
+#define MPL_PROC_MUTEX_POSIX_H_INCLUDED
+
+#include <errno.h>
+#include <pthread.h>
+
+typedef pthread_mutex_t MPL_proc_mutex_t;
+
+#if defined(MPL_NEEDS_PTHREAD_MUTEXATTR_SETTYPE_DECL)
+int pthread_mutexattr_settype(pthread_mutexattr_t * attr, int kind);
+#endif /* MPL_NEEDS_PTHREAD_MUTEXATTR_SETTYPE_DECL */
+
+/* FIXME: convert errors to an MPL_PROC_MUTEX_ERR value */
+
+#ifdef MPL_HAVE_PTHREAD_MUTEXATTR_SETPSHARED
```

Comment:
Where does this get defined? I wasn't able to find it in this PR or in `master`.

Reply:
Same thing for most of the macro definitions in this file.

Reply:
1. `MPL_HAVE_PTHREAD_MUTEXATTR_SETPSHARED` is defined in configure.ac when checking `AC_CHECK_FUNCS(pthread_mutexattr_setpshared)`.

1. `MPL_NEEDS_PTHREAD_MUTEXATTR_SETTYPE_DECL` is defined by `PAC_FUNC_NEEDS_DECL([#include <pthread.h>],pthread_mutexattr_settype)`.

1. `MPL_PPROC_MUTEX_ERRORCHECK_VALUE` is mistake. it should be `MPL_PTHREAD_MUTEX_ERRORCHECK_VALUE` (**I will fix this**). 
Please search definition of `PTHREAD_MUTEX_ERRORCHECK_VALUE` in mpl/configure.ac.

The `AX_PREFIX_CONFIG_H([include/mplconfig.h],[MPL])` line in configure.ac adds the `MPL_` prefix for all defined macros when creating `include/mplconfig.h`. But it does not replace the undef macros comments. That might be the reason you can not find them.